### DISCUSSION TITLE
`MARKER` and its interaction across word lists.

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#177] [General]
I assume that setting `MARKER` (marking a spot in the dictionary and data space) and subsequently jumping around multiple word lists with `SET-CURRENT` between definitions, then invoking that marker will have undefined behaviour.  I've not yet tried this, but I suspect word links across multiple word lists could be broken leaving the system in an undefined state with some words being unreachable.

If word lists are supported it seems that `MARKER` should be removed, since it only seems to make sense in a single dictionary list.

**Comment by ruv** [comment]
The word `marker` was designed to support multiple word lists and multiple address spaces.  It saves the state of each word list, the search order, the pointer of data space, and other system-dependent information that is required to restore the state of the Forth system (in respect to words and word lists). Not sure if it shall save and restore the compilation word list. At least, in some implementation it does that.
